aws-lc-rs/src/aead.rs

```diff
@@ -1169,7 +1137,7 @@ impl TlsRecordSealingKey {
     #[inline]
     #[allow(clippy::needless_pass_by_value)]
     pub fn seal_in_place_append_tag<A, InOut>(
-        &self,
+        &mut self,
```

Comment:
With this, you should now be able to remove the `Mutex` from the struct as well.

Reply:
Blah missed this, thanks!